torch/testing/_internal/common_methods_invocations.py

```diff
@@ -704,6 +704,37 @@ def sample_inputs_index_fill(op_info, device, dtype, requires_grad):
             samples.append(SampleInput((tensor, d, idx_nonctg, fill_val)))
     return samples
 
+def sample_inputs_index_copy(op_info, device, dtype, requires_grad):
+    samples = []
+    t = make_tensor((S, S, S), device, dtype,
+                    low=None, high=None,
+                    requires_grad=requires_grad)
+    s = make_tensor((S, S, S), device, dtype,
+                    low=None, high=None,
+                    requires_grad=requires_grad)
+    # non-contiguous input
+    t01 = t.transpose(0, 1)
+    t02 = t.transpose(0, 2)
+    t12 = t.transpose(1, 2)
+    # non-contiguous input
+    s01 = t.transpose(0, 1)
```

Comment:
This is probably the source of the gradcheck failures. It looks like you mean to use `s.transpose` here? Gradcheck has an oddity where 

```
gradcheck(torch.index_copy, (a, 0, torch.tensor((0, 1)), a.T))
```

will fail, even though computing the gradient in eager mode will succeed. cc @albanD

Reply:
This MAY be the source of the jit failure, too, although the operator is jittable in this case on Colab:

```
def foo(t):
  return torch.index_copy(t, 0, torch.tensor((1, 0)), t.T)
scripted_foo = torch.jit.script(foo)
out = scripted_foo(a)
```

Reply:
You are completely right! Sorry for this, this was my first PR on the C++ side, and after looking up the JIT errors, it seemed like it could have been caused by some internals.